test/quantization/test_quantized_op.py

```diff
@@ -2274,6 +2274,38 @@ def test_empty_batch(self):
         result = torch.ops.quantized.linear_dynamic(X, w_packed)
         self.assertEqual(result.shape, (0, 2))
 
+    def test_advanced_indexing(self):
+        """
+        Verifies that the x[:, [0], :, :] syntax works for quantized tensors.
+        """
+        for dtype in (torch.qint8, torch.quint8, torch.qint32):
+            x_q = torch.quantize_per_tensor(
+                torch.randn(1, 4, 4, 4), 0.1, 0, dtype)
+            # reference
+            x_fp32 = x_q.dequantize()
+
+            # single dim, single index
+            x_q_s1 = x_q[:, [0], :, :]
+            x_fp32_s1 = x_fp32[:, [0], :, :]
+            self.assertTrue(torch.allclose(x_fp32_s1, x_q_s1.dequantize()))
```

Comment:
should we test for equality of x_q_s1 and x_fp32_s1 + quant + dequant with the same qparams? similar for other cases

Reply:
sorry, what do you mean?  Anything different from `self.assertTrue(torch.allclose(x_fp32_s1, x_q_s1.dequantize()))`?

Reply:
oh I mean:
```
self.assertEqual(x_fp32_s1, x_q_s1.dequantize())
```

or as Raghu suggested, we can compare the quantized tensor
```
x_fp32_s1_q = torch.quantize_per_tensor(x_fp32_s1, ...)
self.assertEqual(x_fp32_s1_q, x_q_s1) # this will compare quantized tensor, including qparams and int_repr
```